model/atmosphere/dycore/docs/icon4py_sphinx.py

```diff
@@ -0,0 +1,364 @@
+    var_type_in_inputs = True
+    var_type_formatting = '``'
+    print_variable_longnames = True
+
+    def get_doc(self):
+        # Override the default get_doc method to pick up all docstrings in the
+        # source code
+        source = inspect.getsource(self.object) # this is only the source of the method, not the whole file
+
+        docstrings = self.get_docstring_blocks(source, self.docstring_keyword)
+
+        docstrings_list = []
+        for idocstr, docstring in enumerate(docstrings):
+            formatted_docstr = None
+
+            # Get some useful information on the following method call
+            call_string = self.get_next_method_call(source, source.find(docstring) + len(docstring))
+            next_method_info = self.get_method_info(call_string)
+            # and process a scientific documentation docstring
+            formatted_docstr = docstring.splitlines()
+            formatted_docstr = self.format_docstring_block(formatted_docstr, self.docstring_keyword)
+            formatted_docstr = self.add_header(formatted_docstr, next_method_info)
+            formatted_docstr = self.process_scidocstrlines(formatted_docstr, next_method_info)
+            formatted_docstr = self.add_next_method_call(formatted_docstr, call_string)
+
+            if formatted_docstr is not None:
+                if idocstr < len(docstrings)-1: # Add footer
+                    formatted_docstr = self.add_footer(formatted_docstr)
+                # add the processed docstring to the list
+                docstrings_list.append(formatted_docstr)
+
+        return docstrings_list
+    
+    def get_docstring_blocks(self, source, keyword):
+        # which in this implementation are comment blocks
+        comment_blocks = []
+        source_lines = source.splitlines()
+        in_block = False
+        current_block = []
+        for line in source_lines:
+            stripped_line = line.strip()
+            if stripped_line.startswith(f'# {keyword}:'):
+                if in_block:
+                    comment_blocks.append('\n'.join(current_block))
+                    current_block = []
+                in_block = True
+                current_block.append(line)
+            elif in_block:
+                if stripped_line.startswith('#'):
+                    current_block.append(line)
+                else:
+                    in_block = False
+                    comment_blocks.append('\n'.join(current_block))
+                    current_block = []
+        if current_block:
+            comment_blocks.append('\n'.join(current_block))
+        return comment_blocks
+
+    def format_docstring_block(self, docstr_lines, keyword):
+        # Clean up and format
+        if docstr_lines[0].strip().startswith(f'# {keyword}:'):
+            docstr_lines.pop(0) # remove the {keyword} prefix
+        while docstr_lines[0].strip() == '#':
+            # strip empty lines from the beginning
+            docstr_lines.pop(0)
```

Comment:
This could be skipped if the strings are already cleaned when adding them.

Reply:
see comment above